.github/workflows/test.yml

```diff
@@ -51,15 +53,36 @@ jobs:
     - run: nix-env -iA nix-build-uncached -f nix/
 
     - name: "nix-build"
-      run: nix-build-uncached --max-jobs 4 -A all-systems-go -build-flags -L
+      run: nix-build-uncached --max-jobs auto -A all-systems-go -build-flags -L
+
+    - name: Discover `test/bench` changes
+      if: github.actor != 'dependabot[bot]' && github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request'
+      run: |
+        echo "Branch name: ${{ github.event.pull_request.head.ref }}"
+        echo "Branch here: ${{ github.head_ref || github.ref_name }}"
+        git diff --no-index $(nix-build -A tests.bench)/share test/bench/ok | patch -p1 -R
+        git switch ${{ github.event.pull_request.head.ref }}
+        git branch --show-current
+
+    - name: Commit `test/bench` changes
+      if: github.actor != 'dependabot[bot]' && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request'
+      uses: EndBug/add-and-commit@v9
+      with:
+        default_author: github_actions
+        author_name: Cycle and memory benchmark updater
+        github_token: ${{ secrets.NIV_UPDATER_TOKEN }}
+        committer_name: GitHub Actions
+        committer_email: actions@github.com
+        message: "Updating `test/bench` numbers"
+        commit: -- test/bench/ok
```

Comment:
this is trying to be selective, not sure if it works as expected with a prior `git add`